src/prefect/engine.py

```diff
@@ -1281,14 +1282,26 @@ def link_state_to_result(state: State, result: Any) -> None:
     - The field can be preserved on copy.
     - We cannot set this attribute on Python built-ins.
     """
-    # We cannot track some Python built-ins since they are singletons and could create
-    # confusing relationships, e.g. `None`
-    if type(result) in UNTRACKABLE_TYPES:
-        return
+
+    def link_if_not_untrackable(obj: Any, state: State, ctx: FlowRunContext) -> None:
+        """Track connection between a task run result and its associated state if it has a unique ID.
+
+        We cannot track booleans, Ellipsis, None, NotImplemented, or the integers from -5 to 256
+        because they are singletons.
+        """
+        if (type(obj) in UNTRACKABLE_TYPES) or (
+            type(obj) == int and obj in UNTRACKABLE_INTS
```

Comment:
Why define `UNTRACKABLE_INTS` explicitly rather than do the range check here? 
```suggestion
            isinstance(obj, int) and (-5 <= obj <= 256)
```

Reply:
I have a minor preference for the range check because creating a set of ~250 objects feels a little heavy for checking if the rule is satisfied (I say that only because the rule exists, if it was a bunch of strings we'd have no choice but to create the set). But not a strong opinion, the set check itself is lightweight.